Added way for translation links to be page-specific

gjvnq opened the pull request:
Added file `translation-links.json` which specifies equivalent pages across translations.

So a page like `/en/what-is-gender` can send the user to `/es/que-es-genero` when the user clicks on the Spanish language link.

gjvnq committed:
Added way for translation links to be page-specific

ErdaradunGaztea commented:
Hmm, the only problem I can think of is that all translation-related data is supposed to be stored inside separate folders for each language. I'm more of a R/Python girl, so how feasible would it be to implement in JS something like that?

- Each language has a `translation-links.json` file with a list of `"TAG": "/xx/filename"` entries.
- These files are combined into a map during build process, probably similar to the existing one.

gjvnq committed:
Moved languages list to package.json

Also edited public/*/_strings.js so they all show
the strings in the same order.

gjvnq commented:
If you don't mind making the english version the "main authority", we could
have each page have an `en-equiv` metadata entry use some JavaScript to
stich the whole thing together.

The problem is that  it doesn't allow for two different languages to have
"sister pages" that don't exist in English.

Em seg., 30 de mai. de 2022 12:35, Laura Bąkała ***@***.***>
escreveu:

> Hmm, the only problem I can think of is that all translation-related data
> is supposed to be stored inside separate folders for each language. I'm
> more of a R/Python girl, so how feasible would it be to implement in JS
> something like that?
>
>    - Each language has a translation-links.json file with a list of "TAG":
>    "/xx/filename" entries.
>    - These files are combined into a map during build process, probably
>    similar to the existing one.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GenderDysphoria/GenderDysphoria.fyi/pull/98#issuecomment-1141287003>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAXIYHBFTTREK3QUGJW6TL3VMTN27ANCNFSM5XK4OGZQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
